### ISSUE TITLE
Add and test helper functions for generating uniform random interventions

### ISSUE THREAD

**Issue by eb8680** [BasisResearch/chirho#246]
The design in https://github.com/BasisResearch/chirho/pull/240 and previous iterations includes a number of new components on which the overall correctness of causation and explanation query implementations depends. To reduce the difficulty of reasoning about these implementations and make incremental progress toward landing causation and explanation queries in master, we should pull out these components and specify and test them in isolation.

One of these components is a pair of helper function for generating stochastic interventions whose values are drawn from (approximately) uniform distributions. These can be used with `chirho.counterfactual.ops.split` to cast the search for alternative antecedent values as an inference problem. In the draft design in #240 these functions are called `uniform_proposal` and `random_intervention`:
```python
def uniform_proposal(
    support: pyro.distributions.constraints.Constraint, **kwargs
) -> pyro.distributions.Distribution:
    ...

def random_intervention(
    support: Constraint,
    name: str,
    *,
    make_proposal: Callable[[Constraint], Distribution] = uniform_proposal,
) -> Callable[[torch.Tensor], torch.Tensor]:

    def _random_intervention(value: torch.Tensor) -> torch.Tensor:
        event_shape = value.shape[len(value.shape) - support.event_dim:]
        proposal_dist = make_proposal(support, event_shape=event_shape)
        return pyro.sample(name, proposal_dist)

    return _random_intervention
```
`uniform_proposal` should heuristically construct a high-entropy normalized proposal distribution whose `.support` attribute matches the given `pyro.distributions.constraints.Constraint` object, and should be extensible to new `Constraint` subtypes, perhaps via `functools.singledispatch`:
```python
@functools.singledispatch
def uniform_proposal(
    support: pyro.distributions.constraints.Constraint, **kwargs,
) -> pyro.distributions.Distribution:
    if support is pyro.distributions.constraints.real:
        return pyro.distributions.Normal(0., 1.).mask(False)
    elif support is pyro.distributions.constraints.boolean:
        return pyro.distributions.Bernoulli(logits=torch.zeros(()))
    else:
        tfm = pyro.distributions.transforms.biject_to(support)
        base = uniform_proposal(pyro.distributions.constraints.real, **kwargs)
        return pyro.distributions.TransformedDistribution(base, tfm)
```
This is similar in spirit to `pyro.distributions.ImproperUniform`, but `ImproperUniform` is not normalized and does not implement `sample()` and hence cannot be used as a proposal distribution.
